### ISSUE TITLE
MCL1 perturbation 23~66

### ISSUE THREAD

**Issue by brycestx** [choderalab/perses#448]
Per our conversation during the meeting, I think we should also use MCL1 as a test system for 'validation' because it almost 3x smaller than CDK2. The perturbation I am working with is simple and is located at indices 19 > 13 in [MCL1_ligands.sdf.zip](https://github.com/choderalab/perses/files/1627783/MCL1_ligands.sdf.zip). 
[MCL1_protein_fixed.pdb.zip](https://github.com/choderalab/perses/files/1627785/MCL1_protein_fixed.pdb.zip)
is the protein that I prepared using pdbfixer. 

However after preparing the topology proposal with [NonequilibriumFEPSetup](https://github.com/choderalab/perses/blob/master/perses/dispersed/relative_setup.py#L31), I get an error when trying to build the [HybridTopologyFactory](https://github.com/choderalab/perses/blob/master/perses/annihilation/new_relative.py#L14):
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-17-9565cf892879> in <module>()
      1 snefep = NonequilibriumSwitchingFEP(sfesetup[0], sfesetup[1], sfesetup[2], ncmc_nsteps=1000000, 
      2                                     nsteps_per_iteration=1000, trajectory_directory='mcl1_solvated_011118',
----> 3                                     trajectory_prefix='traj')

~/PycharmProjects/perses/perses/dispersed/relative_setup.py in __init__(self, topology_proposal, pos_old, new_positions, use_dispersion_correction, forward_functions, n_equil_steps, ncmc_nsteps, nsteps_per_iteration, temperature, trajectory_directory, trajectory_prefix, atom_selection, scheduler_address)
    362 
    363         #construct the hybrid topology factory object
--> 364         self._factory = HybridTopologyFactory(topology_proposal, pos_old, new_positions, use_dispersion_correction=use_dispersion_correction)
    365 
    366         #use default functions if none specified

~/PycharmProjects/perses/perses/annihilation/new_relative.py in __init__(self, topology_proposal, current_positions, new_positions, use_dispersion_correction, functions)
    146 
    147         #verify that no constraints are changing over the course of the switching.
--> 148         self._constraint_check_fast()
    149 
    150         #construct dictionary of exceptions in old and new systems

~/PycharmProjects/perses/perses/annihilation/new_relative.py in _constraint_check_fast(self)
    455         for constraint_pair in overlapping_constraints:
    456             if old_system_constraints[constraint_pair] != new_system_constraints[constraint_pair]:
--> 457                 raise ValueError("There is a changing constraint length in this system.")
    458 
    459     def _determine_interaction_group(self, atoms_in_interaction):

ValueError: There is a changing constraint length in this system.
```

I am currently trying to debug this. @pgrinaway can you give it a try on your system and see if you get the same error?

**Comment by jchodera** [comment]
@brycestx : This is weird, since we are currently testing the exact same set of MCL1 ligands from the JACS set as part of the tests I added in https://github.com/choderalab/perses/pull/444

That test only checked the ligand transformations, however. It's possible that there is a bug in the relative topology factor when the protein is include din the system as well. We should add a test for that.

**Comment by brycestx** [comment]
@jchodera I'm getting the error on the solvated system only. The complex system doesn't give me the error.

@pgrinaway were you able to reproduce this at all?

**Comment by jchodera** [comment]
> @jchodera I'm getting the error on the solvated system only. The complex system doesn't give me the error.

@pgrinaway : One big issue I spotted was that the [ligand in solvent `TopologyProposal` construction code](https://github.com/choderalab/perses/blob/master/perses/dispersed/relative_setup.py#L276-L280) doesn't specify the [`old_alchemical_atoms`](https://github.com/choderalab/perses/blob/master/perses/rjmc/topology_proposal.py#L115-L117) argument, which means that *all* atoms are assumed to be alchemically changing. We should fix this throughout any drivers you have.

That argument was added in #420 to fix issues we'd been having with horribly slow constraint changes.

**Comment by pgrinaway** [comment]
>One big issue I spotted was that the ligand in solvent TopologyProposal construction code doesn't specify the old_alchemical_atoms argument, which means that all atoms are assumed to be alchemically changing. We should fix this throughout any drivers you have.

Ah, right--I had written that code before you changed the `TopologyProposal` to contain the `old_alchemical_atoms` argument. You're right, we should update the construction code.

**Comment by brycestx** [comment]
fixed by #450

**Comment by pgrinaway** [comment]
Sounds good
